src/NoteMouseHandler/SelectionMouseHandler.js

```diff
@@ -1,72 +1,71 @@
+  constructor() {
+    super([defaultActionFactory, actionFactory], getCursor, defaultCursor)
   }
 }
 
-function actionFactory(e, ctx) {
+function actionFactory(local, ctx, e) {
   if (e.relatedTarget) {
     return null
   }
 
-  const selectionRect = ctx.selection.getBounds(ctx.transform)
-  const cpos = getLocal(e)
-  const clicked = selectionRect && selectionRect.contains(cpos.x, cpos.y)
-  if (!clicked) {
-    if (e.nativeEvent.detail == 2) {
-      return createNoteAction(ctx)
+  const { selection, transform } = ctx
+  if (selection && selection.enabled) {
+    const selectionRect = selection.getBounds(transform)
+    const clicked = selectionRect.contains(local.x, local.y)
+    if (clicked) {
+      const type = getDragPositionType(local.x - selectionRect.x, selectionRect.width)
+      switch (type) {
+        case "center": return moveSelectionAction(ctx)
+        case "right": return dragSelectionRightEdgeAction(ctx)
+        case "left": return dragSelectionLeftEdgeAction(ctx)
+      }
     }
-    return createSelectionAction(ctx)
   }
-  const type = getDragPositionType(cpos.x - selectionRect.x, selectionRect.width)
-  switch (type) {
-    case "center": return moveSelectionAction(ctx)
-    case "right": return dragSelectionRightEdgeAction(ctx)
-    case "left": return dragSelectionLeftEdgeAction(ctx)
+
+  if (e.nativeEvent.detail == 2) {
+    return createNoteAction(ctx)
   }
+  return createSelectionAction(ctx)
 }
 
-function getCursor(e, ctx) {
-  const loc = getLocal(e)
-  const selectionRect = ctx.selection.getBounds(ctx.transform)
-  const hover = selectionRect && selectionRect.contains(loc.x, loc.y)
+function getCursor(local, ctx) {
+  const selectionRect = ctx.selection.enabled && ctx.selection.getBounds(ctx.transform)
+  const hover = selectionRect && selectionRect.contains(local.x, local.y)
   if (hover) {
-    const type = getDragPositionType(loc.x - selectionRect.x, selectionRect.width)
+    const type = getDragPositionType(local.x - selectionRect.x, selectionRect.width)
     return cursorForPositionType(type)
   }
-  return "crosshair"
+  return defaultCursor
 }
 
 const createSelectionAction = ctx => (onMouseDown, onMouseMove, onMouseUp) => {
   const { selection, quantizer, transform } = ctx
   let startPosition
 
   // 選択範囲外でクリックした場合は選択範囲をリセット
-  onMouseDown(e => {
-    const pos = getLocal(e)
-    startPosition = pos
+  onMouseDown(local => {
+    startPosition = local
     selection.reset()
     // TODO: カーソルを移動
   })
 
-  onMouseMove(e => {
-    const pos = getLocal(e)
-    const rect = Rect.fromPoints(startPosition, pos)
+  onMouseMove(local => {
+    const rect = Rect.fromPoints(startPosition, local)
     selection.resize(rect, quantizer, transform)
   })
 
-  onMouseUp(e => {
-    const pos = getLocal(e)
-    const rect = Rect.fromPoints(startPosition, pos)
-    const events = ctx.getEventsInRect(rect)
+  onMouseUp(() => {
+    const events = ctx.getEventsInSelection()
```

Comment:
__[jshint]__
*  'const' is available in ES6 (use 'esversion

<!-- Comment by [SideCI](https://sideci.com) -->